src/storage-service.ts

```diff
@@ -0,0 +1,84 @@
+/**
+ * Mock implementation of StorageService
+ */
+
+import type { StorageService, CommP, DownloadOptions, SettlementResult, ProofSetId, StorageProvider } from './types.js'
+import { MockUploadTask } from './upload-task.js'
+import { normalizeCommP } from './commp.js'
+
+export class MockStorageService implements StorageService {
+  readonly proofSetId: ProofSetId
+  readonly storageProvider: StorageProvider
+  private _storedData: Map<string, Uint8Array> = new Map()
+
+  constructor(proofSetId: ProofSetId, storageProvider: StorageProvider) {
+    this.proofSetId = proofSetId
+    this.storageProvider = storageProvider
+  }
+
+  upload(data: Uint8Array | ArrayBuffer): MockUploadTask {
+    console.log('[Mock] Starting upload...')
+    const uploadTask = new MockUploadTask(data)
+
+    // Store data for later retrieval (in mock)
+    const storeData = async () => {
+      const commp = await uploadTask.commp()
+      const bytes = data instanceof ArrayBuffer ? new Uint8Array(data) : data
+      this._storedData.set(commp.toString(), bytes)
+    }
+    storeData().catch(console.error)
```

Comment:
I would expect this function to fail if the `storeData` call fails. What is the motivation for decoupling the two calls?

Reply:
In case `storeData` fails, can we make the `uploadTask` Promise reject (if it's a Promise, if it's not, find another way to emit an error)?

Reply:
just throwing a quick mock in here for now :jedi-hand-wave: ignore the details I think, we'll fill it in soon